backend/src/v5/routes/sso/aad/aad.js

```diff
@@ -0,0 +1,165 @@
+		respond(req, res, err);
+	}
+};
+
+const establishRoutes = () => {
+	const router = Router({ mergeParams: true });
+
+	/**
+	* @openapi
+	* /sso/aad/authenticate:
+	*   get:
+	*     description: Redirects the user to Microsoft's authentication page and calls authenticate-post endpoint upon successful authentication
+	*     tags: [Aad]
+	*     operationId: authenticate
+	*/
+	router.get('/authenticate', checkAadConfig, authenticate);
+
+	/**
+	 * @openapi
+	 * /sso/aad/authenticate-post:
+	 *   get:
+	 *     description: Redirects the user to 3D Repo's SSO signup page
+	 *     tags: [Aad]
+	 *     operationId: authenticatePost
+	 */
+	router.get('/authenticate-post', checkAadConfig, authenticatePost);
+
+	/**
+	 * @openapi
+	 * /sso/aad/signup:
+	 *   post:
+	 *     description: Redirects the user to Microsoft's authentication page and calls signup-post endpoint upon successful authentication
+	 *     tags: [Aad]
+	 *     operationId: signup
+	 *     requestBody:
+	 *       content:
+	 *         application/json:
+	 *           schema:
+	 *             type: object
+	 *             required:
+	 *             - username
+	 *             - countryCode
+	 *             - mailListAgreed
+	 *             properties:
+	 *               username:
+	 *                 type: string
+	 *                 description: The username of the user
+	 *                 example: username123
+	 *               countryCode:
+	 *                 type: string
+	 *                 description: The country code of the user
+	 *                 example: GB
+	 *               company:
+	 *                 type: string
+	 *                 description: The company of the user
+	 *                 example: 3D Repo
+	 *               mailListAgreed:
+	 *                 type: boolean
+	 *                 description: Whether the user has signed yp for the latest news and tutorials
+	 *                 example: true
+	 *               captcha:
+	 *                 type: string
+	 *                 description: The reCAPTCHA token generated from the sign up form
+	 *                 example: 5LcN0ysfAAAAAHpnld1tAweI7DKU7dswmwnHWYcB
+	 *     responses:
+	 *       401:
+	 *         $ref: "#/components/responses/notLoggedIn"
+	 */
+	router.post('/signup', checkAadConfig, signup);
+
+	/**
+	 * @openapi
+	 * /sso/aad/signup-post:
+	 *   get:
+	 *     description: Retrieves user email, firstname and lastname from Microsoft Graph API, signs a user up and sends a verification email to the email address provided
+	 *     tags: [Aad]
+	 *     operationId: signup-post
+	 *     responses:
+	 *       400:
+	 *         $ref: "#/components/responses/invalidArguments"
+	 *       200:
+	 *         description: Retrieves user email, firstname and lastname from Microsoft Graph API, signs a user up and sends a verification email to the email address provided
+	 */
+	router.get('/signup-post', checkAadConfig, getUserDetailsAndValidateEmail, validateSignUpData, signupPost);
```

Comment:
put name of route in constant

Reply:
At this state, i don't think we need to do another validation of the payload. we just make sure user email is not taken already, piece together the info and call signup.

We also need to differentiate between a user that already signed up, and a user that already signed with SSO and give different error messages as the UI needs to respond differently